libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt

```diff
@@ -119,6 +129,46 @@ class PredictiveCacheController @JvmOverloads constructor(
         mapListeners[map] = onStyleLoadedListener
     }
 
+    /**
+     * Create cache controllers for a map instance.
+     * Call when a new map instance is available.
+     *
+     * @param map an instance of [MapboxMap]
+     * @param styles a list of Mapbox style URIs to cache.
+     * If no styles are passed current map's style will be cached.
+     * Only styles hosted on Mapbox Services are supported.
+     * Only non-volatile styles will be cached.
+     */
+    @JvmOverloads
+    fun createStyleMapControllers(map: MapboxMap, styles: List<String> = emptyList()) {
+        val tileStore = map.getResourceOptions().tileStore
+        if (tileStore == null) {
+            handleError("TileStore instance not configured for the Map.")
+            return
+        }
+
+        val offlineManager = OfflineManager(map.getResourceOptions())
+
+        mapListeners[map]?.let {
+            removeMapControllers(map)
+        }
+
+        val onStyleLoadedListener = OnStyleLoadedListener {
+            if (styles.isEmpty()) {
+                map.getStyle()?.styleURI?.let { styleURI ->
+                    createMapsController(styleURI, map, offlineManager, tileStore)
+                }
+            } else {
+                styles.forEach { styleURI ->
+                    createMapsController(styleURI, map, offlineManager, tileStore)
+                }
```

Comment:
This is not tied to the current map's style lifecycle so we shouldn't need to wait for the primary style to load to create these controllers.

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/5068/commits/a42a365c81916810a07c3257e47e3694ba04c7cf